src/main/java/uk/gov/hmcts/reform/sendletter/entity/LetterState.java

```diff
@@ -3,5 +3,6 @@
 public enum LetterState {
     Created,
     Uploaded,
+    FailedToUpload,
```

Comment:
Why are we creating this state? If an upload fails it should be left in the Created state where it will be retried, there should not be a state for this.

We should mark letters as stale that have been uploaded to Xerox more than X days ago.

Reply:
Having discussed with Rafal, we think it would be best not to have any new state but instead just have this job generate alerts for letters that are in the Uploaded state more than a threshold time - thoughts?

Reply:
As to why question, same [answer](https://github.com/hmcts/send-letter-service/pull/43#discussion_r177739809) can be applied. As to reporting the stale letters (which is being an `Uploaded` state + threshold) - next pr pending to do that exactly.

On top of first question I can add that not only functionality is kind of ported from producer/consumer, but also adapted to new column `state` which is highly important in the process. Moreover making `is_failed` attribute irrelevant. In case of a need to implement retry logic it has to come through the sprint planning with agreed specs rather than speculating and giving hypothetical suggestions to amend the agreed functionality.

That being said, suggestions are great 👍. Main point of not allowing retry for now is answered earlier - not ram the ftp which continues failing with same documents generating unnecessary volumes.